### ISSUE TITLE
Using a computed property as a relationship

### ISSUE THREAD

**Issue by cpa** [django-json-api/django-rest-framework-json-api#215]
Hello! I'm struggling to include a computed property as a relationship of my model.
I have the following models:

```
class Language(BaseModel):
    iso_code = models.CharField(max_length=2)

class Sentence(BaseModel):
    body = models.TextField()
    language = models.ForeignKey(Language)

class Dataset(BaseModel):
    name = models.CharField(max_length=100)
    description = models.TextField()
    sentences = models.ManyToManyField(Sentence, blank=True, related_name='datasets', related_query_name='dataset')
```

Now, I want my `Dataset` model to have a `Language` relationship: it should be the list of all (distinct) languages in the sentences of my dataset.

I tried to add:

```
    @property
    def get_languages(self, obj):
        return TODO
``` 

in my Dataset model definition and

```
    languages = relations.SerializerMethodResourceRelatedField(
        source='get_languages', many=True, model=Language, read_only=True)
``` 

in my DataSerializer, but to no avail (getting `AttributeError: 'property' object has no attribute 'parent'`). What is the correct way to go? Thanks!

**Comment by jerel** [comment]
@cpa if your function is a property on the model then you don't want `SerializerMethodResourceRelatedField` as it is analogous to DRF's `SerializerMethodField`. You likely want `ResourceRelatedField`.

If this doesn't solve your issue I can reopen.

**Comment by cpa** [comment]
Thanks for the answer!
Turns out that `ResourceRelatedField` doesn't work when `many=True`, which is my use case. See this PR : https://github.com/django-json-api/django-rest-framework-json-api/pull/151

**Comment by jsenecal** [comment]
If you want to work on #151 help yourself :) the original submitter seem to have abandoned the request.

**Comment by jerel** [comment]
Fixed by #223
